### DISCUSSION TITLE
Something isn't right with the design on my site? Is this a bug?

### DISCUSSION THREAD

**Discussion by dualdiagnosishub** [code4recovery/12-step-meeting-list#655] [Q&A] [answered]
Hi

My client installed the 12 Step Meeting List plugin to their site but are having problems with the way that it's displaying currently (see attached screenshot). The LIST/MAPS view selector isn't working correctly?

The link to the site:
https://ddainc.org/meetings/?tsml-day=3&tsml-view=map

<img width="1254" alt="Screenshot 2022-01-27 at 10 15 57" src="https://user-images.githubusercontent.com/77211097/151344388-b8281c51-bc96-400f-8921-90aa1e044138.png">

**Comment by kiyote33** [comment]
Hello @nextlevelwebdesign,
It appears that your client may need to reconfigure their **Google** API key or get a new key that is properly configured for their website.  Or they could choose to use the optional **Mapbox** Token for mapping. Either is a good option! Neither will likely charge them money. Mapbox gives 50,000 free map views / month, Google gives 28,500 free views. 
To sign up for a Mapbox token go to https://www.mapbox.com/signup/. You will only need a valid email address to get a Mapbox token; no credit card is required. After signing up and getting a token, it can be entered in the Mapbox Token settings box on the Meetings/Import & Settings page.
To sign up for a Google Key, go tohttps://developers.google.com/maps/documentation/javascript/get-api-key. It is a much more complex process to get a Google API key and a credit card is required. 
Currently we recommend **Mapbox** , but ultimately it is the client's choice. 
Hope this helps resolve your problem...

**Comment by dualdiagnosishub** [comment]
Hi

The map problem is easily fixed. It’s the problem with the List/Map switch/button not displaying that I can’t seem to work out?

Thanks for trying to help….

From: kiyote33 ***@***.***>
Date: Thursday, 27 January 2022 at 17:29
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Next Level Web Design CIC ***@***.***>, Mention ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] Something isn't right with the design on my site? Is this a bug? (Discussion #655)

Hello @nextlevelwebdesign<https://github.com/nextlevelwebdesign>,
It appears that your client may need to reconfigure their Google API key or get a new key that is properly configured for their website. Or they could choose to use the optional Mapbox Token for mapping. Either is a good option! Neither will likely charge them money. Mapbox gives 50,000 free map views / month, Google gives 28,500 free views.
To sign up for a Mapbox token go to https://www.mapbox.com/signup/. You will only need a valid email address to get a Mapbox token; no credit card is required. After signing up and getting a token, it can be entered in the Mapbox Token settings box on the Meetings/Import & Settings page.
To sign up for a Google Key, go tohttps://developers.google.com/maps/documentation/javascript/get-api-key. It is a much more complex process to get a Google API key and a credit card is required.
Currently we recommend Mapbox , but ultimately it is the client's choice.
Hope this helps resolve your problem...

—
Reply to this email directly, view it on GitHub<https://github.com/code4recovery/12-step-meeting-list/discussions/655#discussioncomment-2060787>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ASNCLWOYRDBYJTVKINAUIHDUYF6G3ANCNFSM5M5RO2NQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by kiyote33** [reply]
What theme are you using? Sometimes we see display issues pop up with certain themes.  Let me know your theme name and I can test it in my sandbox to see if I can replicate the issue and suggest a resolution that works for you.

**Reply by andrewp123-bit** [reply]
@kiyote33  Is this something we can test as well? I'm using the Newspaper theme and can't seem to get the TSML UI to load (the meeting page says this error as soon as I switch from Legacy to TSML UI):
![dda144a12417efe00bfefd3e6199be6b](https://github.com/user-attachments/assets/d47a35f1-3d21-49b3-bd32-181b831b7729)

**Reply by joshreisner** [reply]
hi @andrewp123-bit this looks like it would be a bug in tsml ui based on unexpected shape of your data. whats your site url?

**Reply by andrewp123-bit** [reply]
@joshreisner it's currently hosted on wkup.org/meetings -- I have it on the legacy UI right now but can switch it if that helps.  Thank you for the help!

**Reply by joshreisner** [reply]
hm, yea if you could switch it back to TSML UI that would be great. i loaded your data locally and it seemed ok

**Reply by joshreisner** [reply]
oops actually never mind - i'm able to see the issue locally

**Reply by joshreisner** [reply]
i think i have the fix ready, i'll let you know when it's out: https://github.com/code4recovery/tsml-ui/pull/448

**Reply by joshreisner** [reply]
ok the fix is merged - i think you might try it again now

**Reply by andrewp123-bit** [reply]
You are incredible! You've helped me a few times in the last few days when I was stumped. Everything is up and running, thank you kindly. @joshreisner

**Comment by dualdiagnosishub** [comment]
This is the theme they’re using:
MesmerizeVersion: 1.6.109
By Horea Radu<https://extendthemes.com/>
Thanks…

From: kiyote33 ***@***.***>
Date: Thursday, 27 January 2022 at 20:36
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Next Level Web Design CIC ***@***.***>, Mention ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] Something isn't right with the design on my site? Is this a bug? (Discussion #655)

What theme are you using? Sometimes we see display issues pop up with certain themes. Let me know your theme name and I can test it in my sandbox to see if I can replicate the issue and suggest a resolution that works for you.

—
Reply to this email directly, view it on GitHub<https://github.com/code4recovery/12-step-meeting-list/discussions/655#discussioncomment-2061962>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ASNCLWL2TYJCCLES5RXUMBDUYGUFTANCNFSM5M5RO2NQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by kiyote33** [reply]
Good News - I am able to replicate your issue in my sandbox.  I'll see what I can come up with and get back to you when I can. Probably not today...

**Reply by kiyote33** [reply]
It's definitely theme related as I can't duplicate the issue in any other theme I've tried except for Mesmerize. The code in question uses standard bootstrap to display the list|map button group which is not displaying the toggle buttons. You could look at ensuring that Bootstrap is implemented in your site to resolve this. A better and simpler option might be to embed this React JS shortcode in the meetings page: **[tsml_react]**. This is a new feature being used on select sites which we are getting very positive feedback on. It replaces the legacy front end code.

**Comment by joshreisner** [comment, marked as answer]
To fix the segmented button, can you add custom CSS to your theme? You might try adding
```
#tsml #meetings .controls .btn {
  width: 100%;
}
```

**Comment by dualdiagnosishub** [comment]
That worked. Thanks for your help!

From: Josh Reisner ***@***.***>
Date: Friday, 28 January 2022 at 04:33
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Next Level Web Design CIC ***@***.***>, Mention ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] Something isn't right with the design on my site? Is this a bug? (Discussion #655)

To fix the segmented button, can you add custom CSS to your theme? You might try adding

#tsml #meetings .controls .btn {

  width: 100%;

}

—
Reply to this email directly, view it on GitHub<https://github.com/code4recovery/12-step-meeting-list/discussions/655#discussioncomment-2063845>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ASNCLWMHGOLBSMLAMKGCR5LUYIMAJANCNFSM5M5RO2NQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
